### PR TITLE
[stable/concourse] Add config for the port name

### PR DESCRIPTION
**Pull request by austinbv**
This allows for a bit of automoation when it comes to spinning up aws lbs

*****************************************************************************************

Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the 
history. This will make it easier to identify new changes. The PR will be squashed 
anyways when it is merged. Thanks.

*****************************************************************************************

### CODE REVIEW

**Comment by austinbv** [comment]
I signed the agreement for the foundation now

**Comment by austinbv** [comment]
Looks like circleci failed for some unknown reason. We should try and rerun it.

Thanks for looking at my PR!

File: stable/concourse/README.md [hunk 1]
```diff
@@ -77,7 +77,9 @@ The following tables lists the configurable parameters of the Concourse chart an
 | `concourse.sessionSigningKey` | Concourse Session Signing Private Key | *See [#ssh-keys](#ssh-keys)* |
 | `concourse.workerKey` | Concourse Worker Private Key | *See [#ssh-keys](#ssh-keys)* |
 | `concourse.workerKeyPub` | Concourse Worker Public Key | *See [#ssh-keys](#ssh-keys)* |
+| `concourse.atcPortName` | Name of Concourse ATC listen port | `atc` |
 | `concourse.atcPort` | Concourse ATC listen port | `8080` |
+| `concourse.tsaPortName` | Name of Concourse TSA listen port | `name` |
```
Comment (william-tran) [inline stable/concourse/README.md]: `tsa`

File: stable/concourse/values.yaml [hunk 2]
```diff
@@ -139,11 +139,21 @@ concourse:
   oldEncryptionKey:
   encryptionKey:
 
+  ## Name of ATC listen port.
+  ## Name of ref: https://concourse.ci/architecture.html
```
Comment (william-tran) [inline stable/concourse/values.yaml]: Name of ref?

**Review by william-tran — CHANGES_REQUESTED** [review]
Could you also provide more details on the use case?

**Comment by austinbv** [comment]
@william-tran Sure happy to.  We have some pretty significant build automation and a strong convention that we name all parts of our PODs the same across a few different systems. In this case we rely on the `port.name` to match the `servicename`. We talked about either changing our build automation or attempting this PR and figured the PR may be nicer if anyone else is in our situation. Removing hard coding while applying sensible defaults - we thought - increased portability and utility of the chart.

Hopefully this context helps. This small change should allow us to get concourse up and running!

**Comment by william-tran** [comment]
@austinbv I don't see the port name being configurable in other charts so I'm hesitant to give my thumbs up for this unless there are others that can see this feature being useful for a number of users. I'm still pretty new to k8s and chart maintenance, so I'll defer judgement to someone else with more experience. @viglesiasce @mattfarina ?

**Comment by mattfarina** [comment]
/cc @frodenas @viglesiasce @nexeck since this is also a chart you maintain

cc @unguiculus do you have any guidance on the practices here?

There are cases where others have named ports after service names. For example, see [sentry](https://github.com/kubernetes/charts/blob/cb0060835763bcdd68fb080ab4799f36a1033fa1/stable/sentry/templates/service.yaml). In that case it didn't need to be an extra property but could be derived from the service name automatically.

File: stable/concourse/templates/web-svc.yaml [hunk 3]
```diff
@@ -18,13 +18,13 @@ metadata:
 spec:
   type: {{ .Values.web.service.type }}
   ports:
-    - name: atc
+    - name: {{ .Values.concourse.atcPortName }}
```
Comment (mattfarina) [inline stable/concourse/templates/web-svc.yaml:21]: If you're looking to name the ports in a similar or the same manner to the service would it make sense to base it on the service name without an extra property? Something like what the [sentry chart](https://github.com/kubernetes/charts/blob/cb0060835763bcdd68fb080ab4799f36a1033fa1/stable/sentry/templates/service.yaml#L16) does?

**Review by mattfarina — COMMENTED** [review]
The chart is failing CI because the chart version also needs to be incremented.

File: stable/concourse/templates/web-svc.yaml (hunk 3 above)
Comment (austinbv) [reply stable/concourse/templates/web-svc.yaml:21]: Nice I can update the PR to do something like sentry

**Pull request closed without merging**